include/battle.h

```diff
@@ -675,6 +675,24 @@ struct BattleStruct
         typeArg = gBattleMoves[move].type;                            \
 }
 
+#define GET_MOVE_CALLED(move, targetArg)                              \
+{                                                                     \
+    switch (gBattleMoves[move].effect)                                \
+    {                                                                 \
+    case EFFECT_COPYCAT:                                              \
+    case EFFECT_MIRROR_MOVE:                                          \
+    case EFFECT_MIMIC:                                                \
```

Comment:
I think Mimic doesn't need to be included here. It replaces itself with the copied move, which can't be used until the next turn. The AI should have no problem using the copied move at that point.